sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/ModelHelper.java

```diff
@@ -504,37 +504,40 @@ public static BlobQueryHeaders transformQueryHeaders(HttpHeaders headers) {
     }
 
     public static void validateConditionsNotPresent(BlobRequestConditions requestConditions,
-        EnumSet<BlobRequestConditionProperty> invalidConditions) {
+        EnumSet<BlobRequestConditionProperty> invalidConditions, String operationName, String parameterName) {
         if (requestConditions == null) {
             return;
         }
+        List<String> invalidConditionsFound = new ArrayList<>();
         if (invalidConditions.contains(BlobRequestConditionProperty.LEASE_ID)
             && requestConditions.getLeaseId() != null) {
-            throw LOGGER.logExceptionAsError(new IllegalArgumentException("'leaseId' is not applicable to this API."));
+            invalidConditionsFound.add(BlobRequestConditionProperty.LEASE_ID.toString());
         }
         if (invalidConditions.contains(BlobRequestConditionProperty.TAGS_CONDITIONS)
             && requestConditions.getTagsConditions() != null) {
-            throw LOGGER.logExceptionAsError(new IllegalArgumentException("'tagsConditions' is not applicable to "
-                + "this API."));
+            invalidConditionsFound.add(BlobRequestConditionProperty.TAGS_CONDITIONS.toString());
         }
         if (invalidConditions.contains(BlobRequestConditionProperty.IF_MODIFIED_SINCE)
             && requestConditions.getIfModifiedSince() != null) {
-            throw LOGGER.logExceptionAsError(new IllegalArgumentException("'ifModifiedSince' is not applicable to "
-                + "this API."));
+            invalidConditionsFound.add(BlobRequestConditionProperty.IF_MODIFIED_SINCE.toString());
         }
         if (invalidConditions.contains(BlobRequestConditionProperty.IF_UNMODIFIED_SINCE)
             && requestConditions.getIfUnmodifiedSince() != null) {
-            throw LOGGER.logExceptionAsError(new IllegalArgumentException("'ifUnmodifiedSince' is not applicable to "
-                + "this API."));
+            invalidConditionsFound.add(BlobRequestConditionProperty.IF_UNMODIFIED_SINCE.toString());
         }
         if (invalidConditions.contains(BlobRequestConditionProperty.IF_MATCH)
             && requestConditions.getIfMatch() != null) {
-            throw LOGGER.logExceptionAsError(new IllegalArgumentException("'ifMatch' is not applicable to this API."));
+            invalidConditionsFound.add(BlobRequestConditionProperty.IF_MATCH.toString());
         }
         if (invalidConditions.contains(BlobRequestConditionProperty.IF_NONE_MATCH)
             && requestConditions.getIfNoneMatch() != null) {
-            throw LOGGER.logExceptionAsError(new IllegalArgumentException("'ifNoneMatch' is not applicable to this "
-                + "API."));
+            invalidConditionsFound.add(BlobRequestConditionProperty.IF_NONE_MATCH.toString());
+        }
```

Comment:
How hot of a code path is this? Do we expect the general case for `invalidConditions` to be empty or very full? We may be able to make this a little more readable and get some perf depending on the answer to those questions. Avoiding calling `contains()` so often for a likely empty list is probably useful. IDK if this is more readable to you but it is to me. (gihhub won't let me give a suggested edit here)

```java
for (BlobRequestConditionProperty condition : invalidConditions) {
    switch (condition) {
        case LEASE_ID:
            if (requestConditions.getLeaseId() != null) {
                invalidConditionsFound.add(BlobRequestConditionProperty.LEASE_ID.toString());
            }
            break;
        // other cases
    }
}
```

Reply:
Yeah that does look a lot better.

Reply:
Changed

Reply:
If we care about perf then bitflags could be an option.